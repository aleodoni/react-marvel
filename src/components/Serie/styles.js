import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  /* flex: 1; */
  flex-direction: column;
  align-items: stretch;
  opacity: 0.6;
  transition: opacity 0.2s;

  &:hover {
    opacity: 1;
  }
`;

export const Imagem = styled.div`
  display: flex;

  img {
    height: 150px;
    width: 150px;
    border-radius: 4px;
    margin-right: 10px;
    margin-top: 10px;
  }
`;
