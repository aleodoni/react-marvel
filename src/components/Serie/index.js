import React from 'react';
import PropTypes from 'prop-types';

import { Container, Imagem } from './styles';

export default function Serie({ url, onMouseEnter, onMouseLeave }) {
  return (
    <Container onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
      <Imagem>
        <img src={url} alt="imagem" data-testid="serie-img" />
      </Imagem>
    </Container>
  );
}

Serie.propTypes = {
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
};
