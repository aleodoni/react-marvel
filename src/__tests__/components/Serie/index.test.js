import React from 'react';
import { render, cleanup, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Serie from '../../../components/Serie/index';

afterEach(cleanup);

it('renderiza corretamente', () => {
  function handleMouseEnter() {}

  function handleMouseLeave() {}

  const url = 'http://imagem.jpg';

  const { container } = render(
    <Serie
      url={url}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    />
  );

  expect(container).toMatchSnapshot();
});

it('testa atributos', () => {
  function handleMouseEnter() {}

  function handleMouseLeave() {}

  const url = 'http://imagem.jpg';

  const { container } = render(
    <Serie
      url={url}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    />
  );
  const serieImg = getByTestId(container, 'serie-img');

  expect(serieImg).toHaveAttribute('src', 'http://imagem.jpg');
});
