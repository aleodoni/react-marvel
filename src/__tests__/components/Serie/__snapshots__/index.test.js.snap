// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<div>
  <div
    class="sc-bdVaJa cQLjeV"
  >
    <div
      class="sc-bwzfXH bkhZFH"
    >
      <img
        alt="imagem"
        data-testid="serie-img"
        src="http://imagem.jpg"
      />
    </div>
  </div>
</div>
`;
